FeedStoreChallenge/ManagedCache.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  ManagedCache.swift
+//  FeedStoreChallenge
+//
+//  Created by Joshua Bryson on 9/9/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import CoreData
+
+@objc(ManagedCache)
+internal class ManagedCache: NSManagedObject {
+	@NSManaged internal var timestamp: Date
+	@NSManaged internal var feed: NSOrderedSet
+}
+
+extension ManagedCache {
+	static func deletePreviousCache(in context: NSManagedObjectContext) throws {
+		try find(in: context).map(context.delete).map(context.save)
+	}
+
+	static func newUniqueInstance(in context: NSManagedObjectContext) throws -> ManagedCache {
+		try deletePreviousCache(in: context)
+		return NSEntityDescription.insertNewObject(forEntityName: String(describing: ManagedCache.self), into: context) as! ManagedCache
+	}
+
+	class func find(in context: NSManagedObjectContext) throws -> ManagedCache? {
+		let request = ManagedCache.fetchRequest()
+		request.fetchLimit = 1
+		return try context.fetch(request).first as? ManagedCache
```

Comment:
To slightly increase CoreData performance `request.returnsObjectsAsFaults` can be set to `false`.

According to the [documentation](https://developer.apple.com/documentation/coredata/nsfetchrequest/1506756-returnsobjectsasfaults)
> When you execute a fetch, by default `returnsObjectsAsFaults` is `true`; Core Data fetches the object data for the matching records, fills the row cache with the information, and returns managed object as faults. These faults are managed objects, but all of their property data resides in the row cache until the fault is fired. When the fault is fired, Core Data retrieves the data from the row cache. ... If you need to access the property values from the returned objects (for example, if you iterate over all the objects to calculate the average value of a particular attribute), then it is more efficient to set `returnsObjectsAsFaults` to `false` to avoid the additional overhead.

```suggestion
		let request = ManagedCache.fetchRequest()
		request.fetchLimit = 1
		request.returnsObjectsAsFaults = false
		return try context.fetch(request).first as? ManagedCache
```